dask_jobqueue/flux.py

```diff
@@ -0,0 +1,187 @@
+        return (int(minutes) * 60) + float(seconds)
+
+    # days hours, minutes and seconds, MM:SS.mm
+    elif "-" in timestr and timestr.count(":") == 2:
+        days, rest = timestr.split("-", 1)
+        hours, minutes, seconds = rest.split(":")
+        return (
+            (int(days) * 86400)
+            + (int(hours) * 3600)
+            + (int(minutes) * 60)
+            + float(seconds)
+        )
+
+    # hours, minutes, seconds HH:MM:SS.mm
+    elif timestr.count(":") == 2:
+        hours, minutes, seconds = timestr.split(":")
+        return (int(hours) * 3600) + (int(minutes) * 60) + float(seconds)
+
+    raise ValueError(f"Unrecognized time format {timestr}")
+
+
+class FluxJob(Job):
+    # Override class variables
+    submit_command = "flux batch"
+
+    # This can be updated to flux cancel when release 0.48 been out a while
+    cancel_command = "flux job cancel"
+    config_name = "flux"
+
+    def __init__(
+        self,
+        scheduler=None,
+        name=None,
+        queue=None,
+        project=None,
+        account=None,
+        walltime=None,
+        job_cpu=None,
+        job_mem=None,
+        config_name=None,
+        **base_class_kwargs,
+    ):
+        super().__init__(
+            scheduler=scheduler, name=name, config_name=config_name, **base_class_kwargs
+        )
+        # This would better be shared logic across backends...
+        if queue is None:
+            queue = dask.config.get("jobqueue.%s.queue" % self.config_name)
+        if project is not None:
+            warnings.warn("flux does not have support for project", FutureWarning)
+        if account is not None:
+            warnings.warn("flux does not have support for accounts", FutureWarning)
+        if job_mem is not None:
+            warnings.warn("flux does not have support for memory", FutureWarning)
+
+        if walltime is None:
+            walltime = dask.config.get("jobqueue.%s.walltime" % self.config_name)
+        if job_cpu is None:
+            job_cpu = dask.config.get("jobqueue.%s.job-cpu" % self.config_name)
+
+        header_lines = []
+
+        # Error and output files
+        error = "%s/%s-%%J.err" % (self.log_directory, self.job_name or "worker")
+        output = "%s/%s-%%J.out" % (self.log_directory, self.job_name or "worker")
+
+        # Flux header build
+        if self.job_name is not None:
+            header_lines.append("#flux: --job-name=%s" % self.job_name)
+        if self.log_directory is not None:
+            header_lines.append("#flux: --error %s" % error)
+            header_lines.append("#flux: --output %s" % output)
+
+        if queue is not None:
+            header_lines.append("#flux: --queue=%s" % queue)
+
+        # Init resources, always 1 task
+        header_lines.append("#flux: -n 1")
+        header_lines.append("#flux: --cores-per-slot=%s" % self.worker_cores)
+
+        if walltime is not None:
+            header_lines.append("#flux: -t %s" % timestr2seconds(walltime))
```

Comment:
If I understand correctly, flux takes a walltime directive as an integer (seconds), so I'm not sure why you are bothering with a different format in input and define a timestr2seconds function?

Reply:
Clarification: I don't know the format of `walltime` here, but the Flux `-t, --time-limit` option is in floating point minutes by default (though it can tak a `s`, `m`, `h`, or `d` suffix for explicit seconds, minutes, hours, or days)

Reply:
Okay sounds like I should do:

```python
header_lines.append("#flux: -t {walltime}s")
```

Reply:
Nope that definitely didn't work! Going to keep it for now. 

```
#flux: -t 00:02:00'
```
```console
RuntimeError: Command exited with non-zero exit code.
Exit code: 1
Command:
flux batch /tmp/flux-job-47lcw93z.sh
stdout:

stderr:
flux-batch: ERROR: invalid Flux standard duration
```